src/common/packing.jl

```diff
@@ -0,0 +1,25 @@
+# Packing routines used in direct_minimization and newton algorithms.
+# They pack / unpack sets of planewaves to make them compatible to be used in
+# algorithms from KrylovKit or Optim libraries
+
+pack(φ) = vcat(Base.vec.(φ)...) # TODO as an optimization, do that lazily? See LazyArrays
+function unpacking(φ) where T
```

Comment:
Things which bother me about this:
- The "generator" should return both pack and unpack, that makes it have some natural symmetry.
- I find the names too generic on the one hand, but not specific enough on the oter. Essentially what it does is that it transforms from an array of arrays to a continuous array. `pack_arrays` and `unpack_arrays`?

Reply:
In the end, @antoine-levitt suggests to define two funcions pack_arrays(basis, φ) and unpack_arrays(basis, x), and put them in PlaneWaveBasis.jl !

Reply:
sounds good. I had that thought as well.